### ISSUE TITLE
exec checks fail to see execfail shopt

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#3178]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2093
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
shopt -s execfail # allow exec to fail-through
exec command1 "$@"
exec command2 "$@"
echo "Failed to find either 'command1' or 'command2', check installation"
exit 1
```

#### Here's what shellcheck currently says:
```
Line 3	SC2093: Remove "exec " if script should continue after this command.
```
#### Here's what I wanted or expected to see:

No such error. `execfail` is set so the script will continue if the exec on line 3 fails. Given that `execfail` is set, I think the user understands the consequences. Now, if `execfail` was not set, then yes, then SC2093 should apply.

**Comment by koalaman** [comment]
Makes sense. The warning is now suppressed when `shopt -s execfail` appears anywhere in the script (regardless of context).
